pp3/module/Application/src/Application/Controller/AdminController.php

```diff
@@ -326,6 +343,124 @@ public function nbVersionsAction() {
+            $nbVersionId = $this->params()->fromPost('nbVersionId');
+            $subject = $this->params()->fromPost('subject');
+            $emailText = $this->params()->fromPost('emailText');
+
+            $users = [];
+
+            $plugins = $this->_pluginRepository
+                    ->getPluginsByNetBeansVersion($nbVersionId, $onlyVerified);
+
+            foreach ($plugins as $plugin) {
+                foreach ($plugin->getAuthors() as $author) {
+                    $email = $author->getEmail();
+                    if (!array_key_exists($email, $users)) {
+                        $users[$email] = [
+                            'email' => $email,
+                            'name' => $author->getName(),
+                            'plugins' => []
+                        ];
+                    }
+                    if (!in_array($plugin->getName(), $users[$email]['plugins'])) {
+                        $users[$email]['plugins'][] = $plugin->getName();
+                    }
+                }
+            }
+
+            $successMessage .= sprintf("Generating E-Mails for %d plugins and %d users<br>\n",
+                    count($plugins),
+                    count($users));
+
+            if ($this->params()->fromPost('sendPreview')) {
+                $successMessage .= "Only sending preview email!<br>\n";
+
+                /**
+                 * @var \Application\Entity\User
+                 */
+                $user = $this->_userRepository->find($this->getAuthenticatedUserId());
+                $users = [$user->getEmail() => [
+                    'email' => $user->getEmail(),
+                    'name' => $user->getName(),
+                    'plugins' => ['Demo plugin 1', 'Demo plugin 2']
+                ]];
+            }
+
+            $transport = new Mail\Transport\Sendmail();
+
+            $emailCount = 0;
+            foreach (array_values(($users)) as $entry) {
+                $emailCount++;
+                $list = "<ul>";
+                foreach($entry['plugins'] as $pluginName) {
+                    $list .= sprintf("<li>%s</li>", htmlspecialchars($pluginName));
+                }
+                $list .= "</ul>";
+
+                $mail = new Mail\Message();
+                $mail->setFrom('noreply@netbeans.apache.org', 'NetBeans webmaster');
+                $mail->setSubject($subject);
+                $mail->getHeaders()->addHeader(ContentType::fromString('Content-Type: text/html; charset=utf-8'));
+                $mail->setBody(sprintf($emailText,
+                                htmlspecialchars($entry['name']),
+                                htmlspecialchars($this->getHomeUrl()),
+                                $list
+                ));
+                $mail->addTo($entry['email']);
+                $transport->send($mail);
+            }
+
+            $successMessage .= "E-Mails were successfully send: " . $emailCount;
```

Comment:
Typo. Should read "_E-mails were successfully sent:_"

Reply:
Fixed - thankss